tests/security/ebpf/check_cap_bpf.pm

```diff
@@ -12,17 +12,38 @@ use testapi;
 use utils;
 
 sub run {
-    my $f_tcpdump = '/usr/sbin/tcpdump';
     my $capability = 'cap_bpf';
+    my $f_bpf_test = '/tmp/bpf_test';
 
     select_console 'root-console';
+    # Set 'unprivileged_bpf_disabled' to 1
+    validate_script_output('sysctl kernel.unprivileged_bpf_disabled=1', sub { m/kernel.unprivileged_bpf_disabled = 1/ });
+    validate_script_output("cat /proc/sys/kernel/unprivileged_bpf_disabled", sub { m/1/ });
 
-    # Install packages
-    zypper_call('in libcap-progs tcpdump');
+    # Download the C test code and compile
+    assert_script_run('zypper -n in gcc libcap-progs', timeout => 300);
+    assert_script_run('wget ' . autoinst_url . '/data/ebpf/bpf_test.c');
+    assert_script_run("gcc -o $f_bpf_test bpf_test.c");
+    # BPF system call should succeed with root permission
+    validate_script_output("$f_bpf_test", sub { m/BPF: Success/ });
 
-    assert_script_run("getcap $f_tcpdump");
-    assert_script_run("setcap $capability+eip $f_tcpdump");
-    validate_script_output("getcap $f_tcpdump", sub { m/.*$capability.*/ });
+    # Delete the script file `validate_script_output` just created,
+    # to avoid the permission denied issue in common user.
+    script_run('rm /tmp/script*.sh');
+
+    select_console 'user-console';
+    # BPF system call should failed without root permission and cap_bpf capability
+    validate_script_output($f_bpf_test, sub { m/BPF: Operation not permitted/ });
+
+    # Grant cap_bpf capability to binary file just compiled
+    select_console 'root-console';
+    assert_script_run("getcap $f_bpf_test");
+    assert_script_run("setcap $capability+eip $f_bpf_test");
+    validate_script_output("getcap $f_bpf_test", sub { m/.*$capability.*/ });
+
+    # BPF system call should succeed with cap_bpf capability in common user
+    select_console 'user-console';
+    validate_script_output($f_bpf_test, sub { m/BPF: Success/ });
```

Comment:
I suggest you clean up env again here

Reply:
Updated the code to clean up env again before `select_console 'user-console'`, thanks.

Reply:
Updated the test code to clean up the `/tmp` directory environment after tests.